diff --git a/dtls-hmac.h b/dtls-hmac.h
index 75b22c1..3d82640 100644
--- a/dtls-hmac.h
+++ b/dtls-hmac.h
@@ -20,6 +20,7 @@
 
 #include "tinydtls.h"
 
+#ifndef USE_HW_ACCEL
 #ifdef WITH_SHA256
 /** Aaron D. Gifford's implementation of SHA256
  *  see http://www.aarongifford.com/ */
@@ -45,6 +46,7 @@ dtls_hash_finalize(unsigned char *buf, dtls_hash_t ctx) {
   return DTLS_SHA256_DIGEST_LENGTH;
 }
 #endif /* WITH_SHA256 */
+#endif /* USE_HW_ACCEL */
 
 void dtls_hmac_storage_init(void);
 
